### DISCUSSION TITLE
Is there any sort of "Panel"?

### DISCUSSION THREAD

**Discussion by ItsTheSky** [amwx/FluentAvalonia#595] [Q&A] [answered]
Hi there! I was looking through a WinUI application that uses FluentUI and found a component that I can't find in FluentAvalonia yet, although it could be handy. In the code, it's called a "panel", and it's also called this way in [FluentUI Blazor](https://www.fluentui-blazor.net/Panel).
Here's a small demonstration gif:
![2024-07-21-18-27-37](https://github.com/user-attachments/assets/ec344373-c780-4508-aab7-9a8bb29d4bde)
Thanks for your answers in advance :)

**Comment by Nolan5G** [comment, marked as answer]
@ItsTheSky 

So I suppose the common term for this 'Panel' outside WinUI is "Drawer".  The short answer is no.

From what I've seen, there are a lot of things that look like Drawers that don't have the flexibility that Drawers have in, say, a React UI framework.  SplitView and NavigationView are two that come to mind, but also the Expander control as well.

I've tried prototyping one in my own local  custom Avalonia Control library because the NavigationView in Fluent Avalonia is very opinionated... and my progress stalled once I pooled my favorite features from popular UI implementations and had to plan my implementation.

The biggest design pains are that a good drawer control would be designed around being attached or decorating an existing control in either inline or overlay modes... In addition to all sorts of properties for neatly managing such a control.  It quickly became apparent that it's maybe doable to implement, but it's far from trivial.
